src/caches/extrapolation_caches.jl

```diff
@@ -47,3 +47,258 @@ function alg_cache(alg::RichardsonEuler,u,rate_prototype,uEltypeNoUnits,uBottomE
   step_no = zero(Int)
   RichardsonEulerConstantCache(dtpropose,T,cur_order,work,A,step_no)
 end
+
+@cache mutable struct ExtrapolationMidpointDeuflhardConstantCache{QType} <: OrdinaryDiffEqConstantCache
+  # Values that are mutated
+  Q::Vector{QType} # Storage for stepsize scaling factors. Q[n] contains information for extrapolation order (n + alg.n_min - 1)
+  n_curr::Int64 # Storage for the current extrapolation order
+  n_old::Int64 # Storage for the extrapolation order n_curr before perfom_step! changes the latter
+
+  # Constant values
+  subdividing_sequence::Array{BigInt,1}  # subdividing_sequence[n] is used for the (n -1)th internal discretisation
+  stage_number::Array{Int,1} # Stage_number[n] contains information for extrapolation order (n + alg.n_min - 1)
+  # Weights and Scaling factors for extrapolation operators
+  extrapolation_weights::Array{Rational{BigInt},2}
```

Comment:
Do these need to be Rational? Maybe it doesn't matter all that much, but Bigs are allocating, so it would be worthwhile to profile if the calculations on these actually end up being costly, and whether just having them as Float64 may be numerically stable enough to get the job done. If Float64 is enough, then typing these to make `t` would make sense.

Reply:
I haven't played with Floats here so far.

The good thing about the`BigInt`s is that the weights are converted in `perform_step` to the type of `u`. That allows to adapt the precision of the problem.
I'd suspect that if one uses `Float64` weights and say a`BigFloat` representation for the problem, the approximation error will not get any better than `eps(Float64)`.

Reply:
What about just converting to `u`'s type here?

Reply:
Ah I see -  the caches are recreated for every call of `solve`, right? Yes, then the type of `u` should be an option.
I guess I will set up an experiment to check the stability of the weights!

BTW: Would it be possible to move the weights into the algorithm's `struct`? That way the computation of the tables for every `solve` call would be bypassed..